server/src/main/java/org/elasticsearch/index/mapper/SourceLoader.java

```diff
@@ -82,14 +82,24 @@ public boolean reordersFieldValues() {
         @Override
         public Leaf leaf(LeafReader reader) throws IOException {
             SyntheticFieldLoader.Leaf leaf = loader.leaf(reader);
+            if (leaf.empty()) {
+                return new Leaf() {
```

Comment:
Given this has no state I wonder if it's worth having it as a final instance on Leaf?

Reply:
:+1: